### PR TITLE
feat: fetch all LP user data

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

i.e iterates over all pool AssetIds, and iterates over all AssetId AccountIds to fetch user data.

This allows us to have a proper loading state for "Your Positions".

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

Low, ensure pools still function as before

## Testing

- Loading state for "Your Positions" is now happy
  - Initially skeletons
  - As *some* but not all are loaded, the loaded positions are displayed but skeletons are still concatenated while *others* are still loading
  - Skeletons are gone after all are loaded

This can be tested with throttling to reliably test this in case things are too fast

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

https://github.com/shapeshift/web/assets/17035424/e6971412-8fe3-4e1c-bfda-f3406118b80c

### CODE REVIEW

**Comment by gomesalexandre** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/shapeshift/web/6032?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `develop`
  * **PR #6028** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6028?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #6030** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6030?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #6032** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6032?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/6032?utm_source=stack-comment).

File: src/pages/ThorChainLP/YourPositions.tsx [hunk 1]
```diff
@@ -59,41 +61,43 @@ type PositionButtonProps = {
   assetId: AssetId
   name: string
   opportunityId: string
+  accountId: string
   apy: string
+  userPoolData: any // TODO
```
Comment (gomesalexandre) [inline src/pages/ThorChainLP/YourPositions.tsx:66]: uh oh

Comment (gomesalexandre) [reply src/pages/ThorChainLP/YourPositions.tsx]: ad53095c088656e9bf2a0890c3782335655e94f2

**Review by 0xApotheosis — COMMENTED** [review]
First functional note before I review the code is that this PR causes my RUNE aysm LPs to not show anymore.

**Review by 0xApotheosis — CHANGES_REQUESTED** [review]
Love me some skeletons, though we lose our RUNE asymmetrical pools in the PR.

**Comment by gomesalexandre** [comment]
> Love me some skeletons, though we lose our RUNE asymmetrical pools in the PR.

This should be happy now! 

image

- RUNE fix: https://github.com/shapeshift/web/pull/6032/commits/81b688cd58fa13a60356c53edeffb55e8756a108
- Naive deduplication which was needed following the RUNE fix https://github.com/shapeshift/web/pull/6032/commits/36d2afdf4b978030122d62ddd0120c32f3a00bf4
- Less naive deduplication with multi-account future-proofing: https://github.com/shapeshift/web/pull/6032/commits/62a89176ac64395e66527f2aed47e35e7376cf13

**Review by 0xApotheosis — APPROVED** [review]
I'm now seeing all LP positions 🙏

**Pull request merged**